### PR TITLE
Update logmasking for audit v2 logger

### PR DESCRIPTION
**Pull request by VivekVinushanth**
### Proposed changes in this pull request
With this we are intending to mask only the userId by default and will skip masking for any claims that we define in **[audit.log.loggable_user_claim]** config.

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java [hunk 1]
```diff
@@ -350,4 +356,29 @@ public static String getInitiatorType(String initiator) {
...
+        if (!claimsFilters.isEmpty()) {
+            // Strip leading and trailing whitespace from each string in the list
+            List<String> strippedClaims = new ArrayList<>();
+            for (String claim : claimsFilters) {
+                strippedClaims.add(StringUtils.stripToNull(claim));
+            }
+            return strippedClaims;
+        }
+        return new ArrayList<>();
+        }
```
Comment (somindatommy) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java:383]: Is this a formatting error?

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java [hunk 2]
```diff
@@ -350,4 +356,29 @@ public static String getInitiatorType(String initiator) {
...
+
+        Object configValue = IdentityConfigParser.getInstance().getConfiguration().get(LOGGABLE_USER_CLAIMS);
+        List<String> claimsFilters = new ArrayList<>();
+        if (configValue instanceof ArrayList) {
+            claimsFilters = (ArrayList) configValue;
+        } else if (configValue instanceof String) {
+            claimsFilters.add((String) configValue);
+        }
+        if (!claimsFilters.isEmpty()) {
+            // Strip leading and trailing whitespace from each string in the list
```
Comment (somindatommy) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java:375]: Missing a `.` at the end.

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java [hunk 3]
```diff
@@ -350,4 +356,29 @@ public static String getInitiatorType(String initiator) {
         }
         return LoggerUtils.Initiator.User.toString();
     }
+
+    /**
+     * Get the loggable claim uris.
+     *
+     * @return array of loggable claim uris.
```
Comment (somindatommy) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java:363]: Returns a list.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 4]
```diff
@@ -3637,12 +3637,23 @@
...
-    </LoggableUserClaims>
+    {% if audit.log.loggable_user_claim is defined %}
+        {% if audit.log.loggable_user_claim.claims is defined %}
+            <LoggableUserClaims>
+                {% for claim in audit.log.loggable_user_claim.claims %}
+                    <LoggableUserClaim>{{ claim }}</LoggableUserClaim>
+                {% endfor %}
+            </LoggableUserClaims>
+        {% endif %}
+        {% if audit.log.loggable_user_claim|length > 0 %}
```
Comment (somindatommy) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3648]: Don't this need to be an else if?

Comment (VivekVinushanth) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3648]: Can be. But thought of going with IFs

Comment (piraveena) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3648]: Shall we add  some default claims in default.json?

Comment (VivekVinushanth) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3648]: Currently, the default behaviour, is to only mask userID. Am preferring to keep the same behaviour. If the customer wants, they will add claims to log them.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 5]
```diff
@@ -3637,12 +3637,23 @@
...
+            <LoggableUserClaims>
+                {% for key, value in audit.log.loggable_user_claim.items() %}
+                    {% if key != 'claims' %}
```
Comment (VivekVinushanth) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3651]: We, went with if condition (than if-elseif), is to ensure to respect both format of configs at once. 
For, ex:

```
claims = ["http://wso2.org/claims/username","http://wso2.org/claims/userprincipal"]
claim1="http://wso2.org/claims/groups"
```

**Review by piraveena — APPROVED** [review]

**Pull request merged**